runhouse/servers/http/http_client.py

```diff
@@ -433,54 +436,59 @@ def call_module_method(
             + (f".{method_name}" if method_name else "")
         )
         serialization = serialization or "pickle"
-        res = retry_with_exponential_backoff(session.post)(
-            self._formatted_url(f"{key}/{method_name}"),
-            json=CallParams(
-                data=serialize_data(data, serialization),
-                serialization=serialization,
-                run_name=run_name,
-                stream_logs=stream_logs,
-                save=save,
-                remote=remote,
-            ).dict(),
-            stream=True,
-            headers=rns_client.request_headers(resource_address),
-            auth=self.auth,
-            verify=self.verify,
-        )
-
-        if res.status_code != 200:
-            raise ValueError(
-                f"Error calling {method_name} on server: {res.content.decode()}"
-            )
         error_str = f"Error calling {method_name} on {key} on server"
 
-        # We get back a stream of intermingled log outputs and results (maybe None, maybe error, maybe single result,
-        # maybe a stream of results), so we need to separate these out.
-        result = None
-        res_iter = res.iter_lines(chunk_size=None)
-        # We need to manually iterate through res_iter so we can try/except to bypass a ChunkedEncodingError bug
-        while True:
-            try:
-                responses_json = next(res_iter)
-            except requests.exceptions.ChunkedEncodingError:
-                # Some silly bug in urllib3, see https://github.com/psf/requests/issues/4248
-                continue
-            except StopIteration:
-                break
-            except StopAsyncIteration:
-                break
-
-            resp = json.loads(responses_json)
-            output_type = resp["output_type"]
-            result = handle_response(
-                resp, output_type, error_str, log_formatter=self.log_formatter
+        with ThreadPoolExecutor() as executor:
+            # Run logs request in separate thread. Can start it before because it'll wait 5 seconds for the
```

Comment:
maybe i missed this but where are is the delay for the calls request to begin?

Reply:
There's no delay for that... in the `/logs` endpoint server side, there's a delay where it waits for the `call` to start.